recipe/meta.yaml

```diff
@@ -12,19 +12,23 @@ source:
         # CMake 3.11 breaks FLANN's existing src/cpp/CMakeLists.txt. See
         # https://github.com/mariusmuja/flann/issues/369 for more information.
         - src-cpp-fix-cmake-3.11-build.patch
+        # This serializer workaround for VS 2013 doesn't work correctly in VC9.
+        - remove-serializer-workaround.patch
 
 build:
-    number: 1
+    number: 2
 
 requirements:
     build:
-        - gcc     # [unix]
-        - {{ compiler('cxx') }}  # [win]
-        - hdf5 1.10.1|1.10.1.*
+        - {{ compiler('cxx') }}  # [not osx]
+        - gcc     # [osx]
         - cmake
+    host:
+        - libgcc  # [osx]
+        - hdf5
     run:
-        - libgcc  # [unix]
-        - hdf5 1.10.1|1.10.1.*
+        - libgcc  # [osx]
+        - {{ pin_compatible('hdf5', max_pin='x.x.x') }}
```

Comment:
Don't think we need the `pin_compatible` as that should happen automatically now thanks to PR ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/46 ), but will let @isuruf clarify.

Reply:
Hmm. So, with `pin_compatible`, `conda render recipe` gives me:
```
    run:
        - hdf5 >=1.10.1,<1.10.2.0a0
        - libgcc
```
but without it, I get
```
    run:
        - hdf5
        - libgcc
```
I'm not totally sure about how `conda render` interacts with this stuff, but I thought that was bad. @isuruf / @msarahan?

Reply:
Do `conda render recipe -m /path/to/conda_build_config.yaml`

Reply:
Aha! Thanks @isuruf. Will remove `pin_compatible`.